lib/Dialect/TTKernel/IR/TTKernelOps.cpp

```diff
@@ -17,9 +18,15 @@ namespace mlir::tt::ttkernel {
 
 static bool insideDispatchOpRegion(mlir::Operation *op) {
   mlir::Operation *parentOp = op->getParentOp();
+
   if (dyn_cast_or_null<ttmetal::DispatchOp>(parentOp)) {
     return true;
   }
+
+  if (dyn_cast_or_null<scf::ForOp>(parentOp)) {
+    return true;
+  }
```

Comment:
I wonder if we should just make this a recursive function?  I think it could be possible in the future for `scf::ForOp` to appear in other places.

```c++
static bool insideDispatchOpRegion(mlir::Operation *op) {
  mlir::Region* region = op->getParentRegion();
  if (!region) {
    return false;
  }

  mlir::Operation* parentOp = region->getParentOp();
  assert(parentOp);
  if (mlir::isa<ttmetal::DispatchOp>(parentOp)) {
    return true;
  }
  return insideDispatchOpRegion(parentOp);
}
```

Reply:
It doesn't, we'd have to also add this is a special case when a ttkernel exists as an isolated module for testing/future.

Reply:
Recursive function works for out simple eltwise test. @nsmithtt I don't quite understand last comment, what are we missing with this recursive solution

Reply:
Yes I saw it fail when I removed the check, so now we have a recursive function with special case, so we don't need to add any more checks like for `Scf::ForOp`, somethings similar to what you suggested